osbenchmark/aggregator.py

```diff
@@ -1,39 +1,41 @@
+import os
 from typing import Any, Dict, List, Union
 import uuid
 
 from osbenchmark.metrics import FileTestExecutionStore
 from osbenchmark import metrics, workload, config
+from osbenchmark.utils import io as rio
 
 class Aggregator:
-    def __init__(self, cfg, test_executions_dict):
+    def __init__(self, cfg, test_executions_dict, args):
         self.config = cfg
+        self.args = args
         self.test_executions = test_executions_dict
         self.accumulated_results: Dict[str, Dict[str, List[Any]]] = {}
         self.accumulated_iterations: Dict[str, int] = {}
+        self.statistics = ["throughput", "latency", "service_time", "client_processing_time", "processing_time", "error_rate", "duration"]
+        self.test_store = metrics.test_execution_store(self.config)
+        self.cwd = cfg.opts("node", "benchmark.cwd")
 
-    # count iterations for each operation in the workload
-    def iterations(self) -> None:
+    def count_iterations_for_each_op(self) -> None:
         loaded_workload = workload.load_workload(self.config)
-        for task in loaded_workload.test_procedures:
-            for operation in task.schedule:
-                operation_name = operation.name
-                iterations = operation.iterations or 1
-                self.accumulated_iterations.setdefault(operation_name, 0)
-                self.accumulated_iterations[operation_name] += iterations
-
-    # accumulate metrics for each task from test execution results
-    def results(self, test_execution: Any) -> None:
+        for test_procedure in loaded_workload.test_procedures:
+            if test_procedure.name == self.config.opts("workload", "test_procedure.name"):
```

Comment:
Would be good practice to add error handling here in case test_procedure from the test executions is not found in the loaded_workload.
 
Even though all test iterations might have the same test procedure by this point, it's possible where users might be using a workload (e.g. a custom workload or modified official workload) where a test procedure isn't available.

Reply:
Added this!